### PR TITLE
Fix path artifacts

### LINKED ISSUE #85: Background path artifacts
Typing one letter at a time, and selecting a bunch of them, to apply the same background color, since each letter is a piece/fragment/text node, we are creating one rectangle (path) for each letter, instead of a bigger rectangle, and it shows these artifacts:

image

### PR DESCRIPTION
**Pull request by jperedadnr**
Fixes #85

### CODE REVIEW

**Review by abhinayagarwal — APPROVED** [review]

**Review by dlemmermann — COMMENTED** [review]
Will this also work with semi transparent colors?

**Comment by jperedadnr** [comment]
good point!

Actually, the current approach is "less bad" with semi transparent colors, and the proposed change gets worse for these:
when stroke + fill overlap at the border of the shape, they are blended somehow, and that creates an undesired "border" all around the rectangle.

But if we don't add the stroke, the "border" still happens at the vertical sides of the rectangles, because these are not joined...

File: src/main/java/com/gluonhq/richtext/RichTextAreaSkin.java [hunk 1]
```diff
@@ -291,9 +295,13 @@ private void addBackgroundPathsToLayers(List<IndexRangeColor> backgroundIndexRan
...
-                .collect(Collectors.toList());
-        textBackgroundColorPaths.removeIf(path -> !backgroundPaths.contains(path));
-        textBackgroundColorPaths.addAll(backgroundPaths);
+                .forEach(path -> fillPathMap.merge(path.getFill(), path, (p1, p2) -> {
+                    Path union = (Path) Shape.union(p1, p2);
+                    union.setFill(p1.getFill());
+                    return union;
+                }));
+        textBackgroundColorPaths.removeIf(path -> !fillPathMap.containsValue(path));
+        textBackgroundColorPaths.addAll(fillPathMap.values());
```
Comment (abhinayagarwal) [inline src/main/java/com/gluonhq/richtext/RichTextAreaSkin.java:304]: Could be simplified to one loop:

```
for (IndexRangeColor indexRangeBackground : backgroundIndexRanges) {
    final Path path = new BackgroundColorPath(textFlow.rangeShape(indexRangeBackground.getStart(), indexRangeBackground.getEnd()));
    ...
    fillPathMap.merge(path.getFill(), path, (p1, p2) -> {
        Path union = (Path) Shape.union(p1, p2);
        union.setFill(p1.getFill());
        return union;
    });
}
```

Comment (jperedadnr) [reply src/main/java/com/gluonhq/richtext/RichTextAreaSkin.java:304]: It is already one stream call, so I don't see  a difference?

Comment (jperedadnr) [reply src/main/java/com/gluonhq/richtext/RichTextAreaSkin.java:304]: Sorry, but I still don't see a real difference between:

```
Map<Paint, Path> fillPathMap = new HashMap<>();
for (IndexRangeColor indexRangeBackground : backgroundIndexRanges) {
    final Path path = new BackgroundColorPath(textFlow.rangeShape(indexRangeBackground.getStart(), indexRangeBackground.getEnd()));
    ...
    fillPathMap.merge(path.getFill(), path, (p1, p2) -> {
        Path union = (Path) Shape.union(p1, p2);
        union.setFill(p1.getFill());
        return union;
    });
}
```
and
```
Map<Paint, Path> fillPathMap = new HashMap<>();
backgroundIndexRanges.stream()
                 .map(indexRangeBackground -> {
                      final Path path = new BackgroundColorPath(textFlow.rangeShape(indexRangeBackground.getStart(), indexRangeBackground.getEnd()));
    ...
                      return path;
                 })
               .forEach(path -> fillPathMap.merge(path.getFill(), path, (p1, p2) -> {
                     Path union = (Path) Shape.union(p1, p2);
                     union.setFill(p1.getFill());
                     return union;
                 }));
```

What am I missing?

**Review by abhinayagarwal — APPROVED** [review]

**Pull request merged**